os/arch/arm/src/common/up_allocateheap.c

```diff
@@ -165,7 +165,7 @@ void up_allocate_heap(FAR void **heap_start, size_t *heap_size)
 
 	uintptr_t ubss_start = (uintptr_t)USERSPACE->us_bssstart;
 	uintptr_t ubase = (uintptr_t)USERSPACE->us_bssend;
-	size_t usize = REGION_END - ubase;
+	size_t usize = (uint32_t)__usram_segment_size__ - ((uint32_t)ubase - (uint32_t)__usram_segment_start__);
```

Comment:
Doesn't  __usram_segment_size__ alone give the usize?

Reply:
Also please modify the below  `ubase` validation  accordingly..
`DEBUGASSERT(ubase` < (uintptr_t)REGION_END);`